### PR TITLE
Update bmpi3

### PR DESCRIPTION
**Pull request by correaa**
## Proposed changes

Minimal changes in real space QMCPACK to update the MPI wrapper.
Few changes in AFQMC.

## What type(s) of changes does this code introduce?

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Ubuntu 20.04 with flags `cmake -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx -DQMC_NIO_MAX_SIZE=32 -DQMC_CUDA=0 -DBUILD_PPCONVERT=1 -DBUILD_AFQMC=1 ..`

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
OK to test

**Comment by prckent** [comment]
^^^ to start getting CI

File: src/Message/Communicate.cpp [hunk 1]
```diff
@@ -52,10 +52,10 @@ Communicate::~Communicate()
 //exclusive:  MPI or Serial
 #ifdef HAVE_MPI
 
-Communicate::Communicate(const mpi3::communicator& in_comm) : d_groupid(0), d_ngroups(1), GroupLeaderComm(nullptr)
```
Comment (ye-luo) [inline src/Message/Communicate.cpp:55]: Is this just a temporal workaround?

File: src/Message/Communicate.h [hunk 2]
```diff
@@ -247,7 +247,7 @@ class Communicate : public CommunicatorTraits
 
 #ifdef HAVE_MPI
   /// mpi3 communicator wrapper
-  mpi3::communicator comm;
+  mutable mpi3::communicator comm;
```
Comment (ye-luo) [inline src/Message/Communicate.h:250]: Is this just a temporal workaround?

**Comment by ye-luo** [comment]
CIs are failing for "boost::mpi3::intranode" in the AFQMC code. https://jenkins-ci.cels.anl.gov/job/QMCPACK-CI-GNU-CUDA/1063/label=compute004.mcs.anl.gov,type=Real-Mixed-enable-CUDA/console

File: src/Message/Communicate.h (hunk 2 above)
Comment (correaa) [reply src/Message/Communicate.h]: Seems to be permanent, indicating that `Communicate` is an explicitly non thread-safe class out of the box.

File: src/Message/Communicate.cpp (hunk 1 above)
Comment (correaa) [reply src/Message/Communicate.cpp:55]: `mpi3::communicator const&` cannot perform the operations desired at this point. A non-const version can.

**Comment by correaa** [comment]
I had a problem checkout this branch so I created a new one. I don't understand why this happened. The result is that I had to create a new PR. https://github.com/QMCPACK/qmcpack/pull/2946

**Pull request closed without merging**